arroyo-api/src/connection_tables.rs

```diff
@@ -86,45 +98,235 @@ async fn get_and_validate_connector<E: GenericClient>(
+    responses(
+        (status = 200, description = "Job output as 'text/event-stream'"),
+    ),
+)]
+pub(crate) async fn test_connection_table(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+    WithRejection(Json(req), _): WithRejection<Json<ConnectionTablePost>, ApiError>,
+) -> Result<Sse<impl Stream<Item = Result<Event, Infallible>>>, ErrorResp> {
+    let client = client(&state.pool).await?;
+    let auth_data = authenticate(&state.pool, bearer_auth).await?;
+
+    let (connector, _, config, schema) =
+        get_and_validate_connector(&req, &auth_data, &client).await?;
+
+    let (tx, mut rx) = channel(8);
+
+    let (tx2, rx2) = channel(8);
+
+    connector
+        .test(&req.name, &config, &req.config, schema.as_ref(), tx)
+        .map_err(|e| {
+            Status::invalid_argument(format!("Failed to parse config or schema: {:?}", e))
+        })?;
+
+    tokio::spawn(async move {
+        loop {
+            let m = match rx.recv().await {
+                Some(d) => d,
+                None => {
+                    warn!("Stream closed");
+                    break;
+                }
+            };
+
+            let message = match m {
+                Ok(msg) => msg,
+                Err(_) => {
+                    error!("Stream error");
+                    break;
+                }
+            };
+
+            let e: Result<Event, Infallible> =
+                Ok(Event::default().data(serde_json::to_string(&message).unwrap()));
+            if tx2.send(e).await.is_err() {
+                break;
+            }
+
+            if message.done {
+                break;
+            }
+        }
+    });
+
+    Ok(Sse::new(ReceiverStream::new(rx2)))
+}
+
+fn get_connection_profile(
+    c: &DbConnectionTable,
+    connector: &dyn ErasedConnector,
+) -> Option<ConnectionProfile> {
+    let config = serde_json::to_string(&c.connection_config.as_ref()?).unwrap();
+    Some(ConnectionProfile {
+        id: c.pub_id.clone(),
+        name: c.connection_name.as_ref()?.clone(),
+        connector: c.connection_type.as_ref()?.clone(),
+        description: connector.config_description(&config).ok()?,
+        config,
+    })
+}
+
+pub(crate) async fn get<C: GenericClient>(
+    auth: &AuthData,
+    client: &C,
+) -> Result<Vec<ConnectionTable>, Status> {
+    let tables = api_queries::get_connection_tables()
+        .bind(client, &auth.organization_id)
+        .all()
+        .await
+        .map_err(log_and_map)?;
+
+    let vec: Vec<ConnectionTable> = tables
+        .into_iter()
+        .filter_map(|t| {
+            // return None;
+            let Some(connector) = connector_for_type(&t.connector) else {
+                warn!("invalid connector {} in saved ConnectionTable {}", t.connector, t.id);
+                return None;
+            };
+
+            let connection_profile = get_connection_profile(&t, &*connector);
+
+            let table = serde_json::to_string(&t.config).unwrap();
```

Comment:
this probably should return an error instead of unwrapping (and below as well)—otherwise a single bad config in your database prevents you from using the system

Reply:
`t.config` is now passed directly to `get_schema` as a `serde_json::Value` so no need to unwrap.